pkg/trace/config/config.go

```diff
@@ -101,7 +101,7 @@ type AgentConfig struct {
 	// It maps tag keys to a set of replacements. Only supported in A6.
 	ReplaceTags []*ReplaceRule
 
-	// transaction analytics
+	// transaction analytics, all map keys are lower-cased by Viper.
```

Comment:
Maybe we should enforce it ourselves on top of Viper? I don't like the fact that we depend so much on a side effect of a dependency. If Viper changes its behavior, extracting events for all uppercase names will break but none of our tests will catch it (at least as far as I can tell).